controller/program/tests/tests/rebalance/chain.rs

```diff
@@ -0,0 +1,595 @@
+use crate::{
+    common::SVM,
+    tests::rebalance::test_utils::{
+        add_common_accounts, fixture_lst_state_data, instructions_sysvar, jupsol_wsol_builder,
+        mock_empty_rebalance_record_account, rebalance_ixs, wsol_builder,
+        StartRebalanceKeysBuilder,
+    },
+};
+
+use inf1_ctl_jiminy::{
+    accounts::{
+        pool_state::{PoolState, PoolStatePacked},
+        rebalance_record::RebalanceRecord,
+    },
+    keys::{INSTRUCTIONS_SYSVAR_ID, POOL_STATE_ID, REBALANCE_RECORD_ID},
+};
+
+use inf1_svc_ag_core::inf1_svc_lido_core::solido_legacy_core::TOKENKEG_PROGRAM;
+
+use inf1_test_utils::{
+    fixtures_accounts_opt_cloned, keys_signer_writable_to_metas, mock_mint, mock_token_acc,
+    raw_mint, raw_token_acc, silence_mollusk_logs, upsert_account, PkAccountTup,
+};
+
+use mollusk_svm::{
+    program::keyed_account_for_system_program,
+    result::{Check, InstructionResult, ProgramResult},
+};
+
+use sanctum_spl_token_jiminy::sanctum_spl_token_core::{
+    instructions::transfer::{
+        NewTransferCheckedIxAccsBuilder, TransferCheckedIxData, TRANSFER_CHECKED_IX_IS_SIGNER,
+        TRANSFER_CHECKED_IX_IS_WRITABLE,
+    },
+    state::mint::{Mint, RawMint},
+};
+
+use sanctum_system_jiminy::sanctum_system_core::ID as SYSTEM_PROGRAM_ID;
+
+use solana_account::Account;
+use solana_instruction::Instruction;
+use solana_pubkey::Pubkey;
+
+use std::collections::HashMap;
+
+use proptest::prelude::*;
+
+fn create_transfer_ix(
+    owner: Pubkey,
+    owner_token_account: Pubkey,
+    inp_mint: [u8; 32],
+    inp_mint_decimals: u8,
+    inp_pool_reserves: Pubkey,
+    amount: u64,
+) -> Instruction {
+    let transfer_ix_data = TransferCheckedIxData::new(amount, inp_mint_decimals);
+    let transfer_accs = NewTransferCheckedIxAccsBuilder::start()
+        .with_src(owner_token_account.to_bytes())
+        .with_mint(inp_mint)
+        .with_dst(inp_pool_reserves.to_bytes())
+        .with_auth(owner.to_bytes())
+        .build();
+
+    let keys = [
+        *transfer_accs.src(),
+        *transfer_accs.mint(),
+        *transfer_accs.dst(),
+        *transfer_accs.auth(),
+    ];
+
+    Instruction {
+        program_id: Pubkey::new_from_array(TOKENKEG_PROGRAM),
+        accounts: keys_signer_writable_to_metas(
+            keys.iter(),
```

Comment:
i cant believe we went through such trouble to make a struct-array dual representation and then its just thrown away lol

```suggestion
    Instruction {
        program_id: Pubkey::new_from_array(TOKENKEG_PROGRAM),
        accounts: keys_signer_writable_to_metas(
            transfer_accs.0.iter(),
```